VoxelSniperCore/src/main/java/com/github/kevindagame/VoxelBrushManager.java

```diff
@@ -154,23 +145,29 @@ public int registeredSniperBrushes() {
     /**
      * @return Amount of handles registered with the system under Sniper visibility.
      */
-    public int registeredSniperBrushHandles() {
-        return brushes.size();
+    public long registeredSniperBrushHandles() {
+        return brushes.values().stream().distinct().count();
     }
 
     /**
      * @param clazz Brush class
      * @return All Sniper registered handles for the brush.
      */
     public Set<String> getSniperBrushHandles(Class<? extends IBrush> clazz) {
-        return new HashSet<>(brushes.get(clazz));
+        Set<String> handles = new HashSet<>();
+        for (String key : brushes.keySet()) {
+            if (brushes.get(key).equals(clazz)) {
+                handles.add(key);
+            }
+        }
+        return handles;
     }
 
     /**
      * @return Immutable Multimap copy of all the registered brushes
      */
-    public Multimap<Class<? extends IBrush>, String> getRegisteredBrushesMultimap() {
-        return ImmutableMultimap.copyOf(brushes);
+    public Map<String, Class<? extends IBrush>> getRegisteredBrushesMultimap() {
```

Comment:
```suggestion
    public Map<String, Class<? extends IBrush>> getRegisteredBrushesMap() {
```
If you change that, make sure references are changed too